sql/core/src/test/scala/org/apache/spark/sql/execution/command/v2/DescribeTableSuite.scala

```diff
@@ -0,0 +1,123 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.command.v2
+
+import org.apache.spark.sql.{AnalysisException, Row}
+import org.apache.spark.sql.connector.catalog.TableCatalog
+import org.apache.spark.sql.execution.command
+import org.apache.spark.sql.types.{BooleanType, MetadataBuilder, StringType, StructType}
+import org.apache.spark.util.Utils
+
+/**
+ * The class contains tests for the `DESCRIBE TABLE` command to check V2 table catalogs.
+ */
+class DescribeTableSuite extends command.DescribeTableSuiteBase with CommandSuiteBase {
+  override def namespace: String = "ns1.ns2"
+
+  test("DESCRIBE TABLE using v2 catalog") {
+    withNamespaceAndTable(namespace, "table") { tbl =>
+      spark.sql(s"CREATE TABLE $tbl (id bigint, data string) $defaultUsing" +
+        " PARTITIONED BY (id)")
+      val descriptionDf = spark.sql(s"DESCRIBE TABLE $tbl")
+      assert(descriptionDf.schema.map(field => (field.name, field.dataType)) ===
+        Seq(
+          ("col_name", StringType),
+          ("data_type", StringType),
+          ("comment", StringType)))
+      val description = descriptionDf.collect()
+      assert(description === Seq(
+        Row("data", "string", ""),
+        Row("id", "bigint", ""),
+        Row("", "", ""),
+        Row("# Partitioning", "", ""),
```

Comment:
@cloud-fan this is a WIP PR, but as you can see there are quite a bit of differences between v1 and v2 command. Do you want me to unify the behavior in this PR? If so, should we update v1 to match v2 or the other way around? TIA!

Reply:
yea I know it's going to be hard.

The most difficult part is partitioning. V2 supports `transform` instead of partition column names. I'd suggest especially handling tables with only partition columns, to not surprise the majority of users.
```
a                   	string              	                    
b                   	int                 	                            	                    
# Partition Columns	                    	                                       	                    
b                   	int
```
Once there exists transforms, we use the v2 format
```
a                   	string              	                    
b                   	int      
c                       timestamp           	                            	                    
# Partitioning	                    	                                       	                    
identity(b)
year(c)
```

Reply:
For `Detailed Table Information`, we need to add more information in the v2 command to follow v1, such as `Created Time`. We can match `V1Table` directly first, and then add more built-in v2 table properties later.

@Peng-Lei is EXTERAL a reserved keyword now?

Reply:
> The most difficult part is partitioning. V2 supports `transform` instead of partition column names. I'd suggest specially handling tables with only partition columns, to not surprise the majority of users.

Can I assume that only the `IdentityTransform`s are the partition columns? I wasn't sure if we should apply these to `BucketTransform` and `SortedBucketTransform` since we will be removing some info.

Reply:
Yes, only `IdentityTransforms` represents partition columns